### ISSUE TITLE
Incompatible with typing-extensions 4.6.x

### ISSUE THREAD

**Issue by snarfed** [hashberg-io/typing-validation#1]
Hi! First of all, thank you for building and maintaining all of these IPLD libraries. They're awesome!

You may have already noticed that typing-extensions v4.6.0 made a change that broke this lib. Specifically, this is what I get when I `import bases` with it:

```py
>>> import bases
Traceback (most recent call last):
  File ".../bases/__init__.py", line 40, in <module>
    from . import encoding as encoding
  File ".../bases/encoding/__init__.py", line 293, in <module>
    base8 = FixcharBaseEncoding(alphabet.base8, pad_char="=", padding="include")
  File ".../bases/encoding/fixchar.py", line 106, in __init__
    validate(char_nbits, Union[int, Literal["auto"]])
  File ".../typing_validation/validation.py", line 635, in validate
    _validate_union(val, t)
  File ".../typing_validation/validation.py", line 515, in _validate_union
    validate(val, member_t)
  File ".../typing_validation/validation.py", line 691, in validate
    raise unsupported_type_error
ValueError: Unsupported validation for type typing_extensions.Literal['auto'].
```

Not an emergency. Thanks in advance for looking!

**Comment by MarshalX** [comment]
is not compatible with pydantic v2. any updates?

**Comment by MarshalX** [comment]
bump @sg495

**Comment by declension** [comment]
Blocking us too. Anyone found any workarounds for running Pydantic 2.x on Python <3.11 with this (or `multiformats` more generally)?

**Comment by MarshalX** [comment]
@declension i just write my own and migrated https://github.com/MarshalX/python-libipld

**Comment by declension** [comment]
> @declension i just wrote my own and migrated https://github.com/MarshalX/python-libipld

Thanks - different use case here I think but maybe I'll start working around too then

**Comment by MarshalX** [comment]
> > @declension i just wrote my own and migrated https://github.com/MarshalX/python-libipld
> 
> Thanks - different use case here I think but maybe I'll start working around too then

feel free to contribute more functions that should be bond :)

**Comment by snarfed** [comment]
@sg495 identified the root cause here months ago in https://github.com/hashberg-io/multiformats/issues/10#issuecomment-1581693117 . any chance you all could get a fix for that in sometime soon? thank you in advance!

**Comment by makkus** [comment]
For what its worth, I've been using the code from pull request https://github.com/hashberg-io/typing-validation/pull/2 for the development version of the application I'm working on (Python 3.11, Pydantic >=2.0), and haven't seen any issues so far.

Having a published version that has this fix would be great though.

**Comment by sg495** [comment]
Apologies for the delay in this: personal issues have prevented me from keeping up with this over the past few months. I have now merged the pull request: I will run local tests and publish an updated version shortly.

**Comment by sg495** [comment]
This has now been released on PyPI as [v1.1.0](https://pypi.org/project/typing-validation/1.1.0/).
